### ISSUE TITLE
Duplicate output

### ISSUE THREAD

**Issue by craftzdog** [shubham8550/writer-ai#4]
When I run the command, it outputs duplicate results like so:

<img width="923" alt="Screenshot 2023-04-04 at 12 33 40" src="https://user-images.githubusercontent.com/1332805/229680596-52d607ea-1e4b-448e-8d07-f44460c52691.png">

## Reproduction

Write a note:

```
How to read a file in Python? Can you give me an example in Markdown?
```

Run the plugin command.

**Comment by shubham8550** [comment]
I Think Socket Listner got initialized twice for some reason ill look into it 

Originally I directly wanted to use Dalai as a dependency of the plugin so users don't have to start the server and i don't have to deal with sockets but Inkdrop is not able to install it 

I'll fix socket this socket thing (it might be happening when you restart ink drop  , listener getting reregistered so I just have to make sure when ink drop closes  all socket events get destroyed )

**Comment by shubham8550** [comment]
if you can help me get Dalai to work it will be great

so ill just use something like this
![image](https://user-images.githubusercontent.com/35478603/229689006-b0dfc803-1579-4850-bd9f-29777e457438.png)

req PR :)

**Comment by craftzdog** [comment]
I guess you should unbind the `result` event handler after finishing receiving the response from the AI here:

https://github.com/shubham8550/writer-ai/blob/b746d8d1dd5b9c84abae7e0ab52b7328b70c2868/lib/writer-ai-main.js#L66

**Comment by craftzdog** [comment]
> Originally I directly wanted to use Dalai as a dependency of the plugin so users don't have to start the server and i don't have to deal with sockets but Inkdrop is not able to install it

I don't have much time for looking into it deeper, sorry. (I'm currently busy on [this](https://forum.inkdrop.app/t/got-the-new-version-rejected-by-apple/3863))
But if you give me a specific error or something, maybe I can help you.

**Comment by shubham8550** [comment]
![image](https://user-images.githubusercontent.com/35478603/229789189-76147d52-2221-4591-9767-21275fc03b68.png)

i am getting output for  line 40
```
dalai.installed().then(console.log)
```
so we  can say dalai is successfully imported but
when we try to fire up model with

```
dalai.request({

        model: "alpaca.7B",
        prompt: "The following is a conversation between a boy and a girl:",
        }, (token) => {
        console.log(token);
        })

```

we get that error

**Comment by craftzdog** [comment]
Looks like it requires a NodeJS native module `conpty.node`.
Those native modules should match the ABI version of the Electron.
Can you try running the following command in the plugin root directory?

```
ipm rebuild
```

**Comment by shubham8550** [comment]
> Looks like it requires a NodeJS native module `conpty.node`. Those native modules should match the ABI version of the Electron. Can you try running the following command in the plugin root directory?
> 
> ```
> ipm rebuild
> ```

this didn't work but I tried something else to deal with  this issue now i know a way to make this extension better

ill rewrite this extension with more functionality and proper code  (not mess)

**Comment by craftzdog** [comment]
Yeah, I found that requiring native modules in the renderer process is deprecated in the recent Electron.

Looking forward to seeing the new version!
